commands/files/readerfile.go

```diff
@@ -13,10 +13,12 @@ type ReaderFile struct {
 	fullpath string
 	reader   io.ReadCloser
 	stat     os.FileInfo
+	offset   int64
+	baseInfo ExtraInfo
 }
 
-func NewReaderFile(filename, path string, reader io.ReadCloser, stat os.FileInfo) *ReaderFile {
-	return &ReaderFile{filename, path, reader, stat}
+func NewReaderFile(filename, path, abspath string, reader io.ReadCloser, stat os.FileInfo) *ReaderFile {
```

Comment:
@whyrusleeping  I added absolute path, because I was unsure how FullPath was used in the code base.  Is it acceptable for full path to be an absolute path?  I am unsure how FullPath and "add -r" interact.